lib/record_store/provider/dynect.rb

```diff
@@ -83,15 +87,21 @@ def secrets
     end
 
     def build_from_api(api_record)
+      record_type = api_record.fetch('record_type')
       record = api_record.merge(api_record.fetch('rdata')).slice!('rdata').symbolize_keys
 
-      return if record.fetch(:record_type) == 'SOA'
+      return if record_type == 'SOA'
+
+      case record_type
+      when 'ALIAS'
+        record[:cname] = record.delete(:alias)
+      end
```

Comment:
@sbfaulkner @es I'm thinking we can eliminate this and modify record/alias.rb to use an attribute called `alias` instead of `cname` for consistency.

DNSimple doesn't seem to rely on the property name `cname` with ALIAS records.  Doing this would ensure that we have a consistent naming pattern in managing records, instead of stubbing Record objects after they've been created.